PowerPointLabs/PowerPointLabs/ActionFramework/Common/Extension/CheckBoxActionHandlerExtensions.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using Microsoft.Office.Interop.PowerPoint;
+using Microsoft.Office.Tools;
+using PowerPointLabs.ActionFramework.Common.Interface;
+using PowerPointLabs.Models;
+
+namespace PowerPointLabs.ActionFramework.Common.Extension
+{
+    /// <summary>
+    /// Extend action handler with powerpoint context.
+    /// Type `this` in the class/subclass of CheckBoxActionHandler to access the APIs below.
+    /// </summary>
+    [Obsolete("DO NOT use this class in your feature! Used only by Action Framework.")]
+    static class CheckBoxActionHandlerExtensions
+    {
+#pragma warning disable 0618
+        public static Application GetApplication(this CheckBoxActionHandler handler)
```

Comment:
Can you also do a refactoring:
all (abstract) handler implements the same interface (empty), and then just need one extensions for that interface, instead of adding extensions for every handler